src/main/java/Joey.java

```diff
@@ -0,0 +1,218 @@
+                break;
+            } else if (userCommand.equalsIgnoreCase("list")) {
+                System.out.println(" Here are the tasks in your list:");
+                for (int i = 0; i < taskCount; i++) {
+                    Task task = tasks[i];
+                    System.out.println(" " + (i + 1) + "." + task.getType() + "[" + (task.isDone() ? "X" : " ") + "] " + task.getDescription());
+                }
+                System.out.println("____________________________________________________________");
+            } else if (userCommand.startsWith("todo")) {
+                String description = userCommand.substring(5);
+                tasks[taskCount] = new Todo(description);
+                taskCount++;
+                System.out.println(" Got it. I've added this task:");
+                System.out.println("   " + tasks[taskCount - 1]);
+                System.out.println(" Now you have " + taskCount + " tasks in the list.");
+                System.out.println("____________________________________________________________");
+            } else if (userCommand.startsWith("deadline")) {
+                String[] parts = userCommand.substring(9).split("/by");
+                String description = parts[0].trim();
+                String by = parts[1].trim();
+                tasks[taskCount] = new Deadline(description, by);
+                taskCount++;
+                System.out.println(" Got it. I've added this task:");
+                System.out.println("   " + tasks[taskCount - 1]);
+                System.out.println(" Now you have " + taskCount + " tasks in the list.");
+                System.out.println("____________________________________________________________");
+            } else if (userCommand.startsWith("event")) {
+                String[] parts = userCommand.substring(6).split("/from|/to");
+                String description = parts[0].trim();
+                String from = parts[1].trim();
+                String to = parts[2].trim();
+                tasks[taskCount] = new Event(description, from, to);
+                taskCount++;
+                System.out.println(" Got it. I've added this task:");
+                System.out.println("   " + tasks[taskCount - 1]);
+                System.out.println(" Now you have " + taskCount + " tasks in the list.");
+                System.out.println("____________________________________________________________");
+            } else if (userCommand.startsWith("mark")) {
+                int taskIndex = Integer.parseInt(userCommand.split(" ")[1]) - 1;
+                if (taskIndex >= 0 && taskIndex < taskCount) {
+                    Task task = tasks[taskIndex];
+                    task.markDone();
+                    System.out.println(" Nice! I've marked this task as done:");
+                    System.out.println("   [" + (task.isDone() ? "X" : " ") + "] " + task.getDescription());
```

Comment:
```suggestion
                    System.out.println(task);
```
Suffices to print all the information needed since toString() has been appropriately overridden. Likewise for all subsequent occurrences.